### PR TITLE
Expose all parameters of OSQP solver.

### PR DESCRIPTION
**Pull request by hongkai-dai**
This change is [Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/14886)

### CODE REVIEW

**Review by hongkai-dai — COMMENTED** [review]
+@russtedrake for both reviews please, thanks!

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/14886#-:-MY2T2VcDhIRc4ajILvI:bshz79q)* status: LGTM missing from assignee RussTedrake(platform), needs at least two assigned reviewers (waiting on @RussTedrake)

**Comment by RussTedrake** [comment]
Thanks!  Can we also update the docstring (perhaps with other missing solvers)?

https://drake.mit.edu/pydrake/pydrake.solvers.mathematicalprogram.html?highlight=solveroptions#pydrake.solvers.mathematicalprogram.SolverOptions

and perhaps even reference that SolverOptions structure documentation from here:
https://drake.mit.edu/pydrake/pydrake.solvers.mathematicalprogram.html?highlight=setsolveroptions#pydrake.solvers.mathematicalprogram.MathematicalProgram.SetSolverOption

(it's so hard to find!)

**Review by RussTedrake — COMMENTED** [review]
> Reviewed 1 of 1 files at r1.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/14886)* status: LGTM missing from assignee RussTedrake(platform), needs at least two assigned reviewers (waiting on @hongkai-dai)

**Review by hongkai-dai — COMMENTED** [review]
Done, I added the documentation, and referenced `SolverOptions` in MathematicalProgram::SetSolverOption.

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/14886#-:-MY3MDp_3hVuBqI5X_EF:ba6cn4x)* status: LGTM missing from assignee RussTedrake(platform), needs at least two assigned reviewers (waiting on @RussTedrake)

**Review by hongkai-dai — COMMENTED** [review]
@ericcousineau-tri Now in `documentation_pybind.h`, for `SetSolverOption` function, I got
```
        // Symbol: drake::solvers::MathematicalProgram::SetSolverOption
        struct /* SetSolverOption */ {
          // Source: drake/solvers/mathematical_program.h:2614
          const char* doc_was_unable_to_choose_unambiguous_names =
R"""(See set_solver_option for more details. Set the double-valued options.)""";
        } SetSolverOption;
```
where `SetSolverOption` is overloaded for three different data types (int, double, string). What is the recommended way to resolve this?

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/14886#-:-MY3Y2e_C8lLCeTOjJNM:b-o9aruh)* status: LGTM missing from assignee RussTedrake(platform), needs at least two assigned reviewers (waiting on @RussTedrake)

**Review by RussTedrake — COMMENTED** [review]
https://github.com/RobotLocomotion/drake/blob/a98c62d08ae3d5a59476f473231ffb985a500459/bindings/pydrake/pydrake_doxygen.h#L309

> Reviewed 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/14886#-:-MY4_PSD-5m9s_lxGa-T:bpxownb)* status: LGTM missing from assignee RussTedrake(platform), needs at least two assigned reviewers (waiting on @hongkai-dai)

___
*[solvers/solver_options.h, line 48 at r2](https://reviewable.io/reviews/robotlocomotion/drake/14886#-MY4_WIZ9WHWhT2Lqh9d:-MY4_WIZ9WHWhT2Lqh9e:bkbqvv4) ([raw file](https://github.com/robotlocomotion/drake/blob/41c32ebe609c0390da18d6d00614be94404096a6/solvers/solver_options.h#L48)):*
> ```objc
>  *
>  * "OSQP" -- Parameter name and values as specified in OSQP Reference
>  * https://osqp.org/docs/interfaces/solver_settings.html#solver-settings
> ```

btw, thanks!  are there other solvers that have been added with options that we should list?

**Comment by EricCousineau-TRI** [comment]
To expand on Russ's link from above:
https://github.com/RobotLocomotion/drake/blob/a98c62d08ae3d5a59476f473231ffb985a500459/bindings/pydrake/pydrake_doxygen.h#L309-L318
TL;DR:
`@exclude_from_pydrake_mkdoc{Explanation}` or `@pydrake_mkdoc_identifier{foobar}`

**Review by hongkai-dai — COMMENTED** [review]
Thanks @russtedrake and @ericcousineau-tri . I used `pydrake_mkdoc_identifier`.

> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/14886#-:-MY65rFnAQ1mby2kDFJP:b-g37eci)* status: LGTM missing from assignee RussTedrake(platform), needs at least two assigned reviewers (waiting on @RussTedrake)

___
*[solvers/solver_options.h, line 48 at r2](https://reviewable.io/reviews/robotlocomotion/drake/14886#-MY4_WIZ9WHWhT2Lqh9d:-MY65gyE5N-lAFE6PGmz:befkq7x) ([raw file](https://github.com/robotlocomotion/drake/blob/41c32ebe609c0390da18d6d00614be94404096a6/solvers/solver_options.h#L48)):*
<details><summary><i>Previously, RussTedrake (Russ Tedrake) wrote…</i></summary><blockquote>

btw, thanks!  are there other solvers that have been added with options that we should list? 
</blockquote></details>

I haven't exposed the solver options for the other solvers yet (like CLP), currently we can't set the options for other solvers. I will do that next.

**Comment by hongkai-dai** [comment]
@drake-jenkins-bot linux-bionic-gcc-bazel-experimental-everything-release please

**Review by RussTedrake — COMMENTED** [review]
:lgtm:

> Reviewed 2 of 2 files at r3.
> *[Reviewable](https://reviewable.io/reviews/robotlocomotion/drake/14886#-:-MY7s9Gc9wSyeBwCvDnk:bnfp4nl)* status: needs at least two assigned reviewers (waiting on @hongkai-dai)
